build-libs/get-latest-content-sha-for-product.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+/**
+ * We're using the docs endpoints to fetch the latest SHA, so we use
+ * the env var for the docs API.
+ */
+const MKTG_CONTENT_DOCS_API = process.env.MKTG_CONTENT_DOCS_API
+
+/**
+ * A map of all possible `product` slugs to known content API endpoints that
+ * will return an object with a `ref` property that accurately reflects the
+ * ref from which the latest content was uploaded.
+ */
+const KNOWN_LATEST_REF_ENDPOINTS = {
+	boundary: '/api/content/boundary/nav-data/latest/docs',
+	nomad: '/api/content/nomad/nav-data/latest/docs',
+	vault: '/api/content/vault/nav-data/latest/docs',
+	vagrant: '/api/content/vagrant/nav-data/latest/docs',
+	packer: '/api/content/packer/nav-data/latest/docs',
+	consul: '/api/content/consul/nav-data/latest/docs',
+	'terraform-docs-common':
+		'/api/content/terraform-docs-common/nav-data/latest/docs',
+	'hcp-docs': '/api/content/hcp-docs/nav-data/latest/docs',
+	'ptfe-releases': '/api/content/ptfe-releases/nav-data/latest/enterprise',
+	sentinel: '/api/content/sentinel/nav-data/latest/sentinel',
```

Comment:
Are the other Terraform repos with nav data out of scope for this PR?

Reply:
Great question! I don't think we need to worry about other content source repos in this context, as this is the same set of content sources from which we fetch redirects [upstream](https://github.com/hashicorp/dev-portal/blob/24199bc75838857d31f7a87cafc55df8c9b50af1/build-libs/redirects.js#L117).

Or to put it another way, it seems that all Terraform redirects are authored in either `terraform-docs-common` or `ptfe-releases` (maybe to reduce the risk of collisions?).

Reply:
Got it -- thanks for explaining!